src/Compilers/CSharp/Test/Syntax/Parsing/NullConditionalAssignmentParsingTests.cs

```diff
@@ -0,0 +1,592 @@
+                        N(SyntaxKind.CoalesceExpression);
+                        {
+                            N(SyntaxKind.IdentifierName);
+                            {
+                                N(SyntaxKind.IdentifierToken, "c");
+                            }
+                            N(SyntaxKind.QuestionQuestionToken);
+                            N(SyntaxKind.IdentifierName);
+                            {
+                                N(SyntaxKind.IdentifierToken, "d");
+                            }
+                        }
+                    }
+                }
+                EOF();
+            }
+        }
+
+        [Theory]
+        [InlineData(SyntaxKind.BarEqualsToken)]
+        [InlineData(SyntaxKind.AmpersandEqualsToken)]
+        [InlineData(SyntaxKind.CaretEqualsToken)]
+        [InlineData(SyntaxKind.LessThanLessThanEqualsToken)]
+        [InlineData(SyntaxKind.GreaterThanGreaterThanEqualsToken)]
+        [InlineData(SyntaxKind.GreaterThanGreaterThanGreaterThanEqualsToken)]
+        [InlineData(SyntaxKind.PlusEqualsToken)]
+        [InlineData(SyntaxKind.MinusEqualsToken)]
+        [InlineData(SyntaxKind.AsteriskEqualsToken)]
+        [InlineData(SyntaxKind.SlashEqualsToken)]
+        [InlineData(SyntaxKind.PercentEqualsToken)]
+        [InlineData(SyntaxKind.EqualsToken)]
+        [InlineData(SyntaxKind.QuestionQuestionEqualsToken)]
+        public void VariousAssignmentKinds_LeftMemberAccess(SyntaxKind kind)
+        {
+            string op = SyntaxFacts.GetText(kind);
+            string source = $"a?.b {op} c";
+            UsingExpression(source, TestOptions.Regular13);
+            verify();
+
+            UsingExpression(source, TestOptions.RegularPreview);
+            verify();
+
+            void verify()
+            {
+                N(SyntaxKind.ConditionalAccessExpression);
+                {
+                    N(SyntaxKind.IdentifierName);
+                    {
+                        N(SyntaxKind.IdentifierToken, "a");
+                    }
+                    N(SyntaxKind.QuestionToken);
+                    N(SyntaxFacts.GetAssignmentExpression(kind));
+                    {
+                        N(SyntaxKind.MemberBindingExpression);
+                        {
+                            N(SyntaxKind.DotToken);
+                            N(SyntaxKind.IdentifierName);
+                            {
+                                N(SyntaxKind.IdentifierToken, "b");
+                            }
+                        }
+                        N(kind);
+                        N(SyntaxKind.IdentifierName);
+                        {
+                            N(SyntaxKind.IdentifierToken, "c");
+                        }
+                    }
+                }
+                EOF();
+            }
+        }
+
+        [Fact]
+        public void Parentheses_Assignment_LHS_01()
+        {
+            UsingExpression("(c?.F) = 1");
```

Comment:
>UsingExpression [](http://example.com/codeflow?start=12&length=15)

Consider testing all the remaining tests with `TestOptions.Regular13` as well.

Reply:
Perhaps pass `LanguageVersion` as a parameter in each case.
```csharp
[Theory]
[CombinatorialData]
public void Parentheses_Assignment_LHS_01(
    [CombinatorialValues(LanguageVersion.CSharp13, LanguageVersion.Preview)]
    LanguageVersion languageVersion)
```

Reply:
sure, probably the best way is by a Theory/TheoryData which is shared between all the tests.